apps/genrsa.c

```diff
@@ -143,8 +151,9 @@ int genrsa_main(int argc, char **argv)
     if (out == NULL)
         goto end;
 
-    BIO_printf(bio_err, "Generating RSA private key, %d bit long modulus (%d primes)\n",
-               num, primes);
+    if (!batch)
+        BIO_printf(bio_err, "Generating RSA private key, %d bit long modulus (%d primes)\n",
+                   num, primes);
```

Comment:
`-batch` doesn't mean non-verbose.  If you want to be able to avoid verbosity, or make it non-default, you might want to look into introducing `-verbose` as well.  Either way, I don't think it's right to make this output depend on `-batch`

Reply:
Actually, I just followed what `ca` and `req` do.

Reply:
If you look, you will see that both commands accept both `-batch` and `-verbose`, where the latter is explained like this:

```
=item B<-verbose>

This prints extra details about the operations being performed.

```

Also, it seems that `doc/man1/ca.pod` explains `-batch` in a bit more detail:

```
=item B<-batch>

This sets the batch mode. In this mode no questions will be asked
and all certificates will be certified automatically.

```

That should make it clearer that `-batch` is to avoid *prompting* while `-verbose` is to allow more output.  So the lines above would more suitably be:

```
    if (verbose)
        BIO_printf(bio_err, "Generating RSA private key, %d bit long modulus (%d primes)\n",
                   num, primes);
```

Reply:
I could make that change, but it would change the default behavior.

Reply:
How’s that? If people invoke the command exactly as they did previously, then they get the same result.

Reply:
Sorry... If people say `-batch` and get no output, that's also a change of behaviour from the definition we've had on that option so far.

Reply:
Well no, but we do have `-batch` in `req` and `ca`, working the way I'm arguing.

-----

This discussion is going in circles and I start to get tired of it.